q2_types/_feature_data.py

```diff
@@ -153,6 +162,26 @@ def generator_to_dna_sequences(view, data_dir):
     skbio.io.write(view, format='fasta', into=file)
 
 
+def paired_dna_sequences_to_generator(data_dir):
+    left = skbio.io.read(os.path.join(data_dir, 'left-dna-sequences.fasta'),
+                         format='fasta', constructor=skbio.DNA)
+    right = skbio.io.read(os.path.join(data_dir, 'right-dna-sequences.fasta'),
+                          format='fasta', constructor=skbio.DNA)
+    for lseq, rseq in zip(left, right):
```

Comment:
`zip` will only consume the length of the shortest iterable, so this will ignore any additional sequences in one file but not the other. Can you update this to use `itertools.zip_longest`, and check that `lseq` and `rseq` are not `None`? Otherwise raise a `ValueError` stating there is a different number of sequences in the two files.